Automattic/jetpack
# Sync: Use the Heartbeat package to generate the stats array

#### Changes proposed in this Pull Request:
The Sync package currently calls `Jetpack::get_stat_data()` to obtain the heartbeat stats array. The Sync package needs to be independent of Jetpack, so use the `Heartbeat::generate_stats_array()` method instead. When the Jetpack plugin is installed, this method returns the same data as `Jetpack::get_stat_data()`.

I also fixed some phpcs warnings in `test_class.jetpack-sync-modules-stats.php`.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:

1. Install, activate, and connect Jetpack.
2. Set up your sandbox and Unagi so you can watch sync actions. For more info: PCYsg-lgg-p2
3. SSH in to your remote site and clear the `last_heartbeat` Jetpack option using wp cli:

    `wp jetpack options delete last_heartbeat`

4. Trigger the heartbeat cron job using wp cli:

    `wp cron event run jetpack_v2_heartbeat`

5. You should see the `jetpack_sync_heartbeat_stats` sync action in Unagi. Expand the meta data and verify that the data includes an array with [these elements](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/class.jetpack-heartbeat.php#L75) (version, wp-version, php-version, etc.).

#### Proposed changelog entry for your changes:
* n/a. I don't think these changes need a changelog entry.
